ddtrace/profiling/collector/_memalloc.c

```diff
@@ -34,9 +33,16 @@ static PyObject* object_string = NULL;
 static bool memalloc_enabled = false;
 
 static void
-memalloc_free(void* ctx, void* ptr)
+memalloc_free(void* ctx, void* ptr, PyMemAllocatorDomain domain)
```

Comment:
Would it be possible to drop this parameter `domain` and keep it somewhere for each of the memory allocators? 

If you look at the docs here https://docs.python.org/3/c-api/memory.html#c.PyMemAllocatorEx
user context `void* ctx` is passed to each of `malloc`, `calloc`, `realloc` and `free`. So, we don't need to have an extra parameter here.